api/app/GraphQL/Mutations/UpdateUserSkillRankings.php

```diff
@@ -0,0 +1,92 @@
+        $userSkillRankingInput = $args['userSkillRanking'];
+
+        // execute blocks depending on whether the value at args.userSkillRankingInput.X is non-null
+        if (isset($userSkillRankingInput['topTechnicalSkillsRanked'])) {
+
+            // wipe existing technical top skills ranking
+            $arrayUserSkillId = $userSkillRankingInput['topTechnicalSkillsRanked'];
+            $userSkillsCollection->where('skill.category', 'TECHNICAL')
+                ->each
+                ->update(['top_skills_rank' => null]);
+
+            // set the rankings using the input array of UserSkill ids
+            $rankIterator = 1;
+            foreach ($arrayUserSkillId as $userSkillId) {
+                $userSkillsCollection->where('id', $userSkillId)->each
+                    ->update(['top_skills_rank' => $rankIterator]);
+                $rankIterator++;
+            }
+        }
+
+        if (isset($userSkillRankingInput['topBehaviouralSkillsRanked'])) {
+
+            // wipe existing behavioural top skills ranking
+            $arrayUserSkillId = $userSkillRankingInput['topBehaviouralSkillsRanked'];
+            $userSkillsCollection->where('skill.category', 'BEHAVIOURAL')
+                ->each
+                ->update(['top_skills_rank' => null]);
+
+            $rankIterator = 1;
+            foreach ($arrayUserSkillId as $userSkillId) {
+                $userSkillsCollection->where('id', $userSkillId)->each
+                    ->update(['top_skills_rank' => $rankIterator]);
+                $rankIterator++;
+            }
+        }
+
+        if (isset($userSkillRankingInput['improveTechnicalSkillsRanked'])) {
+
+            // wipe existing technical improve skills ranking
+            $arrayUserSkillId = $userSkillRankingInput['improveTechnicalSkillsRanked'];
+            $userSkillsCollection->where('skill.category', 'TECHNICAL')
+                ->each
+                ->update(['improve_skills_rank' => null]);
+
+            $rankIterator = 1;
+            foreach ($arrayUserSkillId as $userSkillId) {
+                $userSkillsCollection->where('id', $userSkillId)->each
+                    ->update(['improve_skills_rank' => $rankIterator]);
+                $rankIterator++;
+            }
+        }
+
+        if (isset($userSkillRankingInput['improveBehaviouralSkillsRanked'])) {
+
+            // wipe existing behavioural improve skills ranking
+            $arrayUserSkillId = $userSkillRankingInput['improveBehaviouralSkillsRanked'];
+            $userSkillsCollection->where('skill.category', 'BEHAVIOURAL')
+                ->each
+                ->update(['improve_skills_rank' => null]);
+
+            $rankIterator = 1;
+            foreach ($arrayUserSkillId as $userSkillId) {
+                $userSkillsCollection->where('id', $userSkillId)->each
+                    ->update(['improve_skills_rank' => $rankIterator]);
+                $rankIterator++;
+            }
+        }
```

Comment:
If we do keep this approach...
You're running essentially the same code 4 times. Try putting it in a function. I think the function would take 4 args: $userSkillsCollection, $arrayUserSkillId, $category and $column (top_skills_rank/improve_skills_rank).

Reply:
Okey, done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/7678/commits/d3a41084d287930370066275049438204a3c6bd7